test/functional/readpreference.test.js

```diff
@@ -78,6 +78,50 @@ describe('ReadPreference', function () {
     });
   });
 
+  describe('fromOptions factory method', () => {
+    const { PRIMARY, SECONDARY } = ReadPreference;
+    const TAGS = [{ loc: 'dc' }];
+
+    it('should return undefined if no options are passed', () => {
+      const readPreference = ReadPreference.fromOptions();
+      expect(readPreference).to.be.undefined;
+    });
+
+    context('readPreference is string', () => {
+      it('should accept { readPreference }', function () {
+        const readPreference = ReadPreference.fromOptions({
+          readPreference: PRIMARY
+        });
+        expect(readPreference).to.be.an.instanceOf(ReadPreference);
+        expect(readPreference.mode).to.equal(PRIMARY);
+      });
+
+      it('should accept { readPreference, readPreferenceTags }', function () {
+        const readPreference = ReadPreference.fromOptions({
+          readPreference: SECONDARY,
+          readPreferenceTags: TAGS
+        });
+        expect(readPreference).to.be.an.instanceOf(ReadPreference);
+        expect(readPreference.mode).to.equal(SECONDARY);
+        expect(readPreference.tags).to.eql(TAGS);
+      });
+
+      it('should accept { readPreference, readPreferenceTags, hedge }', function () {
+        const readPreference = ReadPreference.fromOptions({
+          readPreference: SECONDARY,
+          readPreferenceTags: TAGS,
+          hedge: {
+            enabled: true
+          }
+        });
+        expect(readPreference).to.be.an.instanceOf(ReadPreference);
+        expect(readPreference.mode).to.equal(SECONDARY);
+        expect(readPreference.tags).to.eql(TAGS);
+        expect(readPreference.hedge).to.eql({ enabled: true });
+      });
+    });
```

Comment:
It is possible that `primary` and `hedge` options were being combined and since hedge was ignored there was no issue. Now its possible to get a thrown error from the constructor: `'Primary read preference cannot be combined with hedge'`. Can we add a test expecting that thrown error.

Also if you make the other change with the maxStalenessSeconds, we would want a check for that thrown error as well.

Reply:
I wasn't sure how thoroughly to test the factory method.  We have a comprehensive suite of tests for the constructor above the tests I added, and specifically on line 32 we do check for the `'Primary read preference cannot be combined with hedge'` error.  I can add more thorough testing around valid/invalid combinations of options, but I wasn't sure if that was necessary since we basically just reformat the options passed in and defer to the constructor (which is already tested).

Reply:
Yea its a bit of duplication but since now this method throws it where it didn't before I don't want the behavior to go unnoticed so to speak. If we test for it, future changes to the method will be sure to either maintain the behavior or require a intentional choice to modify a test.

I don't think we need any more combinations beyond:
```
expect(() => ReadPreference.fromOptions({ readPreference: 'primary', hedge: { enabled: true } }).to.throw('...')
```

Also [L32](https://github.com/mongodb/node-mongodb-native/blob/NODE-3705/enable-hedge-options-when-readPreference-is-string/test/functional/readpreference.test.js#L32) is a test header for me, I'm actually not seeing a test case for hedge and primary in the constructor section either 🤔

Reply:
I think you were right, I didn't notice that test.  I added this particular test scenario to both the constructor tests and the factory method tests.